vowpalwabbit/example.cc

```diff
@@ -306,6 +307,73 @@ void return_multiple_example(VW::workspace& all, v_array<example*>& examples)
   examples.clear();
 }
 
+namespace model_utils
+{
+size_t read_model_field(io_buf& io, polylabel& poly)
```

Comment:
This sort of introduces a parallel and incompatible way of serializing labels (`label_parser.cache-label`), It might be good to consider if we can reuse that instead